9.traverseTreeAlphabetical.js

```diff
@@ -18,15 +18,26 @@
 //     this.right = null;
 //   }
 // }
-// 
+//
 // f is the root, so therefore
-// traverseTreeAlphabetical(f) Should return the array 
+// traverseTreeAlphabetical(f) Should return the array
 // [ "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K" ]
 
 // Implement your code here
-const traverseTreeAlphabetical = (node, result = []) => {
 
+//DFS traversal - all the way down then up step by step
+
+const traverseTreeAlphabetical = (node, result = []) => {
+  if (!node) return null;
+  let tree = [node];
+  while(tree.length) {
+    let current = tree.shift()
+    result.push(current.val)
+    if (current.left) tree.push(current.left)
+    if (current.right) tree.push(current.right)
 }
+return result
 
+};
```

Comment:
This is a BFS search. For this problem you need to do a DFS search specifically an in order traversal. Please revisit this type of search.